### DISCUSSION TITLE
percent encoding with warp

### DISCUSSION THREAD

**Discussion by ghost** [framesurge/perseus#226] [Q&A]
I compiled [examples/core/custom_server/src/main.rs](https://github.com/framesurge/perseus/blob/main/examples/core/custom_server/src/main.rs) if I try to visit "127.0.0.1:8080/api/echo/hello world", it displayed "hello%20world". How to handle percent encoding?

**Comment by lukechu10** [comment]
Please take a read here https://github.com/seanmonstar/warp/issues/242
This is not really a perseus issue so if you still have problems, it might be worth asking on warp's repo.

**Comment by arctic-hen7** [comment]
Try a library like `urlencoding`.
